CMakeLists.txt

```diff
@@ -31,8 +31,8 @@ target_include_directories(sphinxsys_core INTERFACE ${Simbody_INCLUDE_DIR})
 target_link_directories(sphinxsys_core INTERFACE ${Simbody_LIB_DIR})
 target_link_libraries(sphinxsys_core INTERFACE ${Simbody_LIBRARIES})
 
-find_package(Eigen3 3.3 REQUIRED NO_MODULE)
 if(TARGET Eigen3::Eigen)
+    find_package(Eigen3 3.3 REQUIRED NO_MODULE)
```

Comment:
This effectively prevents to find Eigen installed elsewhere, ok considering the broken current state. The repository was not compiling before that, which means the version bundled with the CI of Eigen is not sufficient. Features used here means it requires at least Eigen 3.4